packages/react-components/react-table/src/components/TableHeaderCell/TableHeaderCell.types.ts

```diff
@@ -10,7 +9,7 @@ export type TableHeaderCellSlots = {
   /**
    * Button handles correct narration and interactions for sorting;
    */
-  button: NonNullable<Slot<ARIAButtonSlotProps>>;
+  button: NonNullable<Slot<'div'>>;
```

Comment:
yes, but it's also a bug fix for an a11y bug with an SLA. Our ARIAButton types always expect `button` to be a valid option. However in this case we never want this slot to be rendered with a `button` because we want the slot to lose all button semantics when the table is not sortable and `role="presentation"` is not a valid role for HTML buttons.

The other option would be for the slot to be rendered conditionally, what would be a bigger change for our partners? 🤔🤔